### ISSUE TITLE
Apply field validator to custom fields

### ISSUE THREAD

**Issue by eronlloyd** [netbox-community/netbox#609]
I'm experimenting with custom fields on devices to create a code to help locate devices in our facilities. Once a convention is defined for custom fields, it would be great to be able to do input validation on them to make sure they comply. It seems possible to create validators, such as a RegexValidator, and apply it during form.clean() to handle something like this.

**Comment by darcynz** [comment]
Being able to apply input validation (with regex) would be great for custom fields but also fields in general. Would help minimize input error at the source.

**Comment by tomehb** [comment]
I strongly agree with this; we are just started populating a Netbox instance with our DC, IPAM and Circuit data and have already noticed a few data errors that don't conform with our internal structures. 

A regex input validation would be an excellent start. Still, taking this further, it would be great to allow administrators to write rules that can use basic logic combined with data within fields and custom fields relating to the current record.

An example for the circuit object, when adding decommissioned date I would want not this before the installed date.

_If  Date_installed > Date_Decommissioned_Custom,
Then true,
Else false_

**Comment by jeremystretch** [comment]
Bumping this from v2.9. More research and planning is needed regarding our overall strategy for custom fields moving forward.

**Comment by jeremystretch** [comment]
Blocked by #4878

**Comment by jeremystretch** [comment]
The implementation of this feature will be limited to providing regular expression support, and likely minimum/maximum values for integer custom fields.

**Comment by ge-fa** [comment]
@jeremystretch Thanks for implementing this, and sorry for this noise in this old, closed issue.

Just a small question: Is it currently possible, to do input validation based on regexes on fields like 'Name'? If not, is something like this planned? Or should I rather open a new issue to discuss this?
